check-update-security.sh

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env bash
+
+set -eio pipefail
+
+usage() {
+    echo "Usage:"
+    echo "  $0 AMI_ID"
+    echo "Example:"
+    echo "  $0 ami-0abc1d2ef34ghij56"
+}
+
+error_msg() {
+    local msg="$1"
+    echo "ERROR: $msg"
+}
+
+#in case of failure, terminate instance
+failure_cleanup() {
```

Comment:
General Qs:
1. Is this script designed to only be run against ECS optimized AMIs?
1. Have we tested on AL1, AL2 (amd64 and arm64), and AL2023 (amd64 and arm64)?
2. Have we tested the clean-up work as expected if any step failed with an error?
3. AL2023 locks repo version (find details [here](https://docs.aws.amazon.com/linux/al2023/ug/deterministic-upgrades.html)), can we confirm this script will work on AL2023? 

Thanks.

Reply:
1. The script has been changed to take in platforms and run against only ECS optimized AMIs
2. We have tested AL2 and AL2023 with both platforms, with older versions showing updates, and newer versions not indicating updates.
3. The clean up works as expected in any failure case. Since we exit on any failure, and we trap the ERR signal with the cleanup function, we run the clean up. In the cases where we exit without failure (due to timeout detections), we run cleanup before exiting. 
4. A new addition has been made to accommodate for the different command necessary with AL2023